src/netbase.cpp

```diff
@@ -42,6 +42,53 @@ bool fNameLookup = DEFAULT_NAME_LOOKUP;
 int g_socks5_recv_timeout = 20 * 1000;
 static std::atomic<bool> interruptSocks5Recv(false);
 
+std::vector<CNetAddr> WrappedGetAddrInfo(const std::string& name, bool allow_lookup)
+{
+    struct addrinfo aiHint;
+    memset(&aiHint, 0, sizeof(struct addrinfo));
+
+    // We want a TCP port, which is a streaming socket type
+    aiHint.ai_socktype = SOCK_STREAM;
+    aiHint.ai_protocol = IPPROTO_TCP;
+    // We don't care which address family (IPv4 or IPv6) is returned
+    aiHint.ai_family = AF_UNSPEC;
+    // If we allow lookups of hostnames, use the AI_ADDRCONFIG flag to only
+    // return addresses whose family we have an address configured for.
+    //
+    // If we don't allow lookups, then use the AI_NUMERICHOST flag for
+    // getaddrinfo to only decode numerical network addresses and suppress
+    // hostname lookups.
+    aiHint.ai_flags = allow_lookup ? AI_ADDRCONFIG : AI_NUMERICHOST;
+
+    struct addrinfo* aiRes = nullptr;
+    int nErr = getaddrinfo(name.c_str(), nullptr, &aiHint, &aiRes);
+    if (nErr) {
```

Comment:
proposed comment (and maybe simplification, as we're not using `nErr` for anything, otherwise if we keep `nErr` it can be const)
```diff
-    int nErr = getaddrinfo(name.c_str(), nullptr, &aiHint, &aiRes);
-    if (nErr) {
+    // getaddrinfo(3) returns 0 if it succeeds or a non-zero error code.
+    if (getaddrinfo(name.c_str(), nullptr, &aiHint, &aiRes)) {
```